### ISSUE TITLE
Allow method rule to grab methods by parameter type

### ISSUE THREAD

**Issue by Nahuel92** [TNG/ArchUnit#996]
Hi there!!
I have to say that this library is awesome, thanks for creating it.
Something I have been trying to do is to create a rule to match certain methods that receive a specific parameter type, but I couldn't make it work because the `haveRawParameterTypes()` is not helpful as it requires an exact match:

```
class Example {
          void someMethod(String stringParam, Integer intParam) {...}
      }
```

`someMethod(String stringParam, Integer intParam)` is matched by `haveRawParameterTypes(String.class, Integer.class)` but I want to match methods by any parameter, not all of them as follows:

```
class Example {
          void someMethod2(String stringParam, Integer intParam, InputStream is) {...}
      }
```

I would like to match `someMethod2(String stringParam, Integer intParam, InputStream is)` by any of the following ways:
- `haveRawParameterTypes(String.class)`.
- `haveRawParameterTypes(Integer.class)`.
- `haveRawParameterTypes(InputStream.class)`.

The motivation behind this is to be able to enforce a certain annotation to be present in methods that receive specific parameter types such as `InputStream`, no matter the order and/or if such methods receive other parameters.
Basically, if a method receives an `InputStream` I want to enforce an annotation to be present, I don't mind if it happens for that method to receive anything else or the parameter order.

Do you know if this behavior is possible or if it will be added in the future?
I hope this description was clear enough, thanks.

**Comment by hankem** [comment]
Does the the following custom `ArchCondition` solve your issue?
```java
static ArchCondition<? super JavaCodeUnit> haveSomeParameterOfRawType(Class<?> expectedType) {
    return new ArchCondition<>("have some parameter of raw type " + expectedType.getName()) {
        @Override
        public void check(JavaCodeUnit codeUnit, ConditionEvents events) {
            long count = codeUnit.getRawParameterTypes().stream()
                    .filter(type -> type.isEquivalentTo(expectedType))
                    .count();
            String message = ConditionEvent.createMessage(codeUnit,
                    "has " + count + " parameters of type " + expectedType.getName()
            );
            events.add(new SimpleConditionEvent(codeUnit, count > 0, message));
        }
    };
}
```
(It can be used in, e.g., `methods().should(haveSomeParameterOfRawType(String.class))`.)

**Comment by hankem** [comment]
If you have `type.isAssignableTo(expectedType)`, you might not even need `type.isEquivalentTo(expectedType)` any more, right? 🤔

**Comment by Nahuel92** [comment]
Sorry, I have copied an old version of my change, but my previous comment is updated now.
